docs/source/how_does_it_work.rst

```diff
@@ -18,59 +18,25 @@ No matter of via the Python API (:ref:`api`), the command line client (:ref:`cmd
-
-    For example the following ``javacc`` code
-
-    .. code-block::
-
-        SqlNode SqlShowTables() :
-        {
-            final Span s;
-            final SqlIdentifier schema;
-        }
-        {
-            <SHOW> { s = span(); } <TABLES> <FROM>
-            schema = CompoundIdentifier()
-            {
-                return new SqlShowTables(s.end(this), schema);
-            }
-        }
-
-    describes a parser line, which understands SQL statements such as
-
-    .. code-block:: sql
-
-        SHOW TABLES FROM "schema"
-
-    While parsing the SQL, they are turned into an instance of the Java class :class:`SqlShowTables` (which is also defined in this project).
-    The :class:`Span` is used internally in Apache Calcite to store the position in the parsed SQL statement (e.g. for better error output).
-    The ``SqlShowTables`` javacc function (not the Java class SqlShowTables) is listed in ``config.fmpp`` as a ``statementParserMethods``, which makes it parsable as main SQL statement (similar to any normal ``SELECT ...`` statement).
-    All Java classes used as parser return values inherit from the Calcite class :class:`SqlNode` or any derived subclass (if it makes sense). Those classes are barely containers to store the information from the parsed SQL statements (such as the schema name in the example above) and do not have any business logic by themselves.
+This function will first give the SQL string to DataFusion via the ``PyO3`` library.
+Inside this class, Apache Arrow DataFusion is used to first parse the SQL string and then turn it into a relational algebra.
+For this, DataFusion uses the SQL language description specified in the `sqlparser-rs library <https://github.com/sqlparser-rs/sqlparser-rs/>`_
+We also include `SQL extensions specific to Dask-SQL <https://github.com/dask-contrib/dask-sql/blob/main/dask_planner/src/parser.rs/>`_. They specify custom language features, such as the ``CREATE MODEL`` statement.
 
 3. SQL is (maybe) optimized
 ---------------------------
 
-Once the SQL string is parsed into an instance of a :class:`SqlNode` (or a subclass of it), Apache Calcite can convert it into a relational algebra and optimize it. As this is only implemented for Calcite-own classes (and not for the custom classes such as :class:`SqlCreateModel`) this conversion and optimization is not triggered for all SQL statements (have a look into :func:`Context._get_ral`).
+Once the SQL string is parsed into an instance of a :class:`SqlNode` (or a subclass of it), DataFusion can convert it into a relational algebra and optimize it. As this is only implemented for DataFusion supported syntax (and not for the custom syntax such as :class:`SqlCreateModel`) this conversion and optimization is not triggered for all SQL statements (have a look into :func:`Context._get_ral`).
```

Comment:
I will create a PR against this PR with some updates here. I am working on this now.

Reply:
Actually, probably easier just to make suggestions here.

Reply:
> Actually, probably easier just to make suggestions here.

That did not go well so I created a PR instead:

https://github.com/randerzander/dask-sql/pull/1

Reply:
thanks for reviewing and completing this, Andy :)